src/main/java/com/sprint/mission/discodeit/service/basic/BasicMessageService.java

```diff
@@ -22,30 +25,62 @@
     private final MessageRepository messageRepository;
+    private final UserRepository userRepository;
     private final BinaryContentRepository binaryContentRepository;
+    private final ChannelRepository channelRepository;
 
     @Override
-    public void messageSaveWithContents(MessageWithContents messageParam, BinaryContent binaryContent) {
-        if (binaryContent == null) {
-            log.info("파일을 하나라도 등록하세요.");
-            return;
+    public Message messageSave(UUID senderId,Message message) {
+        if (message.getContent().trim().isEmpty()) {
+            log.info("메세지 내용을 입력해주세요.");
+            throw new IllegalArgumentException("메세지 내용을 입력해주세요.");
+        }
+        User user = userRepository.findById(senderId).orElseThrow(()->new IllegalArgumentException("사용자를 찾을 수 없습니다."));
+        Channel channel = channelRepository.findById(message.getChannelId()).orElseThrow(()->new IllegalArgumentException("채널을 찾을 수 없습니다."));
+
+        Participant findParticipant=null;
+        for (Participant participant : channel.getParticipants()) {
+            if (participant.getUser().getId().equals(senderId)) {
+                findParticipant = participant;
+                break;
+            }
         }
-        binaryContentRepository.save(binaryContent);
-        List<MultipartFile> files = new ArrayList<>();
-        files.add((MultipartFile) binaryContent);
-        Message message = new Message(messageParam.getContent(), messageParam.getSenderId(), messageParam.getChannelId());
-        message.getMessageFiles().addAll(files);
-        messageRepository.createMessage(message.getId(), message);
+        if(findParticipant == null) {
+            throw new IllegalStateException("채널에 가입하지 않았습니다.");
+        }
+        findParticipant.setMessage(message);
+        return messageRepository.createMessage(message);
     }
+
     @Override
-    public void messageSave(Message message) {
+    public Message messageSaveWithContents(UUID senderId,Message message, List<BinaryContent> files) {
         if (message.getContent().trim().isEmpty()) {
-            log.info("메세지 내용을 입력해주세요.");
-            return;
+            throw new IllegalArgumentException("메세지 내용을 입력해주세요.");
         }
-        messageRepository.createMessage(message.getId(), message);
+        User user = userRepository.findById(senderId).orElseThrow(()->new IllegalArgumentException("사용자를 찾을 수 없습니다."));
+        Channel channel = channelRepository.findById(message.getChannelId()).orElseThrow(()->new IllegalArgumentException("채널을 찾을 수 없습니다."));
+
+        Participant findParticipant=null;
+        for (Participant participant : channel.getParticipants()) {
+            if (participant.getUser().getId().equals(senderId)) {
+                findParticipant = participant;
+                break;
+            }
+        }
+        if(findParticipant == null) {
+            throw new IllegalStateException("채널에 가입하지 않았습니다.");
+        }
```

Comment:
stream을 사용하면 메서드가 더욱 간결해질 것 같아요.  

```suggestion
        Participant findParticipant = channel.getParticipants().stream()
            .filter(participant -> participant.getUser().getId().equals(senderId))
            .findFirst()
            .orElseThrow(() -> new IllegalStateException("채널에 가입하지 않았습니다."));
```